src/Confluent.Kafka/Impl/SafeKafkaHandle.cs

```diff
@@ -258,9 +258,9 @@ internal WatermarkOffsets GetWatermarkOffsets(string topic, int partition)
             return new WatermarkOffsets(low, high);
         }
 
-        internal void Subscribe(ICollection<string> topics)
+        internal void Subscribe(IEnumerable<string> topics)
         {
-            IntPtr list = LibRdKafka.topic_partition_list_new((IntPtr) topics.Count);
+            IntPtr list = LibRdKafka.topic_partition_list_new((IntPtr) topics.Count());
```

Comment:
The infinite problem I stated was only "in general". And if someone provide an infinite enumerable in the parameters, I would say it's his problem ;) 

IEnumerable is the standard for collection (new List<T>(IEnumerable<T>) for example. If you try to create a List from an infinite IEnumerable, it will just hang. So standard way =D